docs/contributing.md

```diff
@@ -0,0 +1,165 @@
+[Discussion](https://github.com/google/jax/discussions)
+to seek feedback on your planned contribution.
+
+## Contributing code using Pull Requests
+
+We do all of our development using git, so basic knowledge is assumed.
+
+Follow these steps to contribute code:
+
+1. Fork the JAX repository by clicking the 'Fork' button on the
+   [repository page](http://www.github.com/google/jax). This create a copy
+   of the Flax repository in your own account.
+
+2. Install Python >=3.6 locally in order to run tests.
+
+3. `pip` installing your fork from source. This allows you to modify the code
+   and immediately test it out:
+
+   ```bash
+   git clone https://github.com/YOUR_USERNAME/jax
+   cd jax
+   pip install -r requirements.txt  # Installs all testing requirements.
+   pip install -e .  # Installs JAX from the current directory in editable mode.
+   ```
+
+4. Add the JAX repo as an upstream remote, so you can use it to sync your
+   changes.
+
+   ```bash
+   git remote add upstream http://www.github.com/google/jax
+   ```
+
+5. Create a branch where you will develop from:
+
+   ```bash
+   git checkout -b name-of-change
+   ```
+
+   And implement your changes using your favorite editor (we recommend
+   [Visual Studio Code](https://code.visualstudio.com/)).
+
+6. Make sure the tests pass by running the following command from the top of
+   the repository:
+
+   ```bash
+   pytest -n auto tests/
+   ```
+
+   JAX's test suite is quite large, so if you know the specific test file that covers your
+   changes, you can limit the tests to that; for example:
+
+   ```bash
+   pytest -n auto tests/lax_scipy_test.py
+   ```
+
+   JAX also offers more fine-grained control over which particular tests are run;
+   see {ref}`running-tests` for more information.
+
+7. Once you are satisfied with your change, create a commit as follows (
+   [how to write a commit message](https://chris.beams.io/posts/git-commit/)):
+
+   ```bash
+   git add file1.py file2.py ...
+   git commit -m "Your commit message"
+   ```
+
+   Then sync your code with the main repo:
+
+   ```bash
+   git fetch upstream
+   git rebase upstream/master
+   ```
+
+   Finally push your commit on your development branch and create a remote 
+   branch in your fork that you can use to create a Pull Request from:
+
+   ```bash
+   git push --set-upstream origin name-of-change
+   ```
+
+8. Create a Pull Request from the JAX repository and send it for review.
+   Check the {ref}`pr-checklist` for considerations when preparing your PR, and
+   consult [GitHub Help](https://help.github.com/articles/about-pull-requests/)
+   if you need more information on using pull requests.
+
+(pr-checklist)=
+
+## JAX Pull Request Checklist
+
+As you prepare a JAX pull request, here are a few things to keep in mind:
+
+### Contributor License Agreement
+
+Contributions to this project must be accompanied by a Contributor License
+Agreement. You (or your employer) retain the copyright to your contribution;
```

Comment:
Suggestion: add `Google` before `Contributor` and the abbreviation in brackets—`(CLA)`— after `Agreement`.